### ISSUE TITLE
How to optimize the insider parameters of the camera?

### ISSUE THREAD

**Issue by HelloMaroon** [facebookresearch/pytorch3d#767]
I'm doing some research about this tutorial:

'https://github.com/facebookresearch/pytorch3d/blob/master/docs/tutorials/camera_position_optimization_with_differentiable_rendering.ipynb'

In this tutorial, I changed the reference image and the mesh object. Then I realize the 'up=((0,0,1),)' in the function 'look_at_view_transform' will influence the rendered image, so I added three parameters to the camera. I just changed a little in the code just like this:

'self.camera_position = nn.Parameter(
            torch.from_numpy(np.array([3,50,0,0,0,1], dtype=np.float32)).to(meshes.device))'

'R, T = look_at_view_transform(self.camera_position[0],self.camera_position[1],self.camera_position[2],
                                      up=((self.camera_position[3],self.camera_position[4],self.camera_position[5]),), device=self.device)'

However, I found that the 'up' didn't change at all, just like this:

Parameter containing:
tensor([ 3.0500, 49.9500,  0.0500,  0.0000,  0.0000,  1.0000], device='cuda:0',
       requires_grad=True)
Parameter containing:
tensor([ 3.5405, 49.4634,  0.5570,  0.0000,  0.0000,  1.0000], device='cuda:0',
       requires_grad=True)
Parameter containing:
tensor([ 3.9828, 49.0348,  1.0663,  0.0000,  0.0000,  1.0000], device='cuda:0',
       requires_grad=True)
Parameter containing:
tensor([ 4.3586, 48.6981,  1.5647,  0.0000,  0.0000,  1.0000], device='cuda:0',
       requires_grad=True)

I wondered that is it available for us to optimize the 'up' with the MSE loss of two images.
Or did I code it with wrong way? Waiting for your reply. Thanks.

**Comment by bottler** [comment]
I think the tuple could be causing problems. Can you try replacing
```
R, T = look_at_view_transform(self.camera_position[0],self.camera_position[1],self.camera_position[2],
up=((self.camera_position[3],self.camera_position[4],self.camera_position[5]),), device=self.device)'
```
with
```
R, T = look_at_view_transform(self.camera_position[0],self.camera_position[1],self.camera_position[2],
up=self.camera_position[None,3:], device=self.device)'
```
?

**Comment by HelloMaroon** [comment]
Thanks a lot, it's solved.
